packages/mdc-select/foundation.ts

```diff
@@ -111,9 +121,11 @@ export class MDCSelectFoundation extends MDCFoundation<MDCSelectAdapter> {
     const previouslySelectedIndex = this.selectedIndex_;
     this.selectedIndex_ = index;
 
-    this.adapter_.setSelectedText(
-        this.selectedIndex_ === numbers.UNSET_INDEX ?
-            '' : this.adapter_.getMenuItemTextAtIndex(this.selectedIndex_).trim());
+    if (this.selectedIndex_ === numbers.UNSET_INDEX || !this.adapter_.getMenuItemTextAtIndex(this.selectedIndex_)) {
+      this.adapter_.setSelectedText('');
+    } else {
+      this.adapter_.setSelectedText(this.adapter_.getMenuItemTextAtIndex(this.selectedIndex_)!.trim());
```

Comment:
nit: Save `this.adapter_.getMenuItemTextAtIndex(this.selectedIndex_)` return value to variable and reuse here and above.

Reply:
I'm gonna keep it like this so it's executed after `this.selectedIndex_ === numbers.UNSET_INDEX` is confirmed false